--- build-n-run-networked.sh	2012-05-04 18:15:03.594327687 +0700
+++ build-n-run.sh-netw-patch	2012-05-04 18:06:25.000000000 +0700
@@ -14,7 +14,7 @@
     exit -1
 fi
 
-$ARTLIBGEN $ARTROOT/src/artlibgen/templates/posix-gcc-mt-file-lint.xml art.h art.c &&
+$ARTLIBGEN $ARTROOT/src/artlibgen/templates/posix-gcc-mt-netw-lint.xml art.h art.c &&
 gcc -c art.c -o art.o -g -ggdb -Wno-pointer-to-int-cast
 
 passOK=0
@@ -22,13 +22,19 @@
 FAILEDlist=""
 total=0
 
+killall -9 artrepgen
+
 for i in `ls [0-9][0-9][0-9].c`; do
 #    set -x
     gcc -W -Wall -Wextra -ansi -g -ggdb -include art.h $i art.o -o $i.out
+    $ARTREPGEN --sock 12345 > tmp &
+    ./usleep 2
     ./$i.out
-    $ARTREPGEN --file tracefile.out > tmp
-#    exit
+    ./usleep 2
+    killall -9 artrepgen
     sed -r 's/[0-9A-Z]{16}//g' tmp > $i.artrep.real
+#    exit
+    mv tmp tracefile.out
     rm -f tmp
     diff -u $i.artrep.real $i.artrep.right
     if test $? -eq 0; then
@@ -56,14 +62,17 @@
 # А теперь для особых случаев
 
 rm -f art.[cho]
-$ARTLIBGEN $ARTROOT/src/artlibgen/templates/posix-gcc-mt-file-special.xml art.h art.c &&
+$ARTLIBGEN $ARTROOT/src/artlibgen/templates/posix-gcc-mt-netw-special.xml art.h art.c &&
 gcc -c art.c -o art.o -g -Wno-pointer-to-int-cast &&
 
 for i in `ls f[0-9][0-9].c`; do
     gcc -W -Wall -Wextra -ansi -g f04-api.c -c
     gcc -W -Wall -Wextra -ansi -g -include art.h $i art.o f04-api.o -o $i.out -I.
+    $ARTREPGEN --sock 12345 > tmp &
+    ./usleep 2
     ./$i.out
-    $ARTREPGEN --file tracefile.out > tmp
+    ./usleep 2
+    killall -9 artrepgen
     sed -r 's/[0-9A-Z]{16}//g' tmp > $i.artrep.real
     rm -f tmp
     diff -u $i.artrep.real $i.artrep.right
