--- build-n-run-tracefile.sh	2021-05-18 21:59:02.853426428 +0700
+++ build-n-run.sh-netw-patch	2021-05-18 22:14:12.661426428 +0700
@@ -2,6 +2,20 @@
 
 SED=$(type gsed > /dev/null && echo gsed || echo sed)
 
+if ! [ -z "$CI_JOB_ID" ]; then
+    # We are inside GitLab CI, setting other variables
+    repGenStartupTime=16
+    repGenShutdownTime=16
+else
+    # Generic run on developer machine
+    repGenStartupTime=3
+    repGenShutdownTime=1
+fi
+echo "repGenStartupTime='$repGenStartupTime'"
+echo "repGenShutdownTime='$repGenShutdownTime'"
+
+killall -9 artrepgen || true
+
 passOK=0
 passFAILED=0
 FAILEDlist=""
@@ -11,15 +25,22 @@
 #    pushd . > /dev/null
     d=`echo $d | ${SED} 's/\.\///g' -`
     if [ "$d" = 019 ]; then
-        RT_TEMPLATE=posix-gcc-mt-file-special
+        RT_TEMPLATE=posix-gcc-mt-netw-special
     else
-        RT_TEMPLATE=""
+        RT_TEMPLATE=posix-gcc-mt-netw-lint
     fi
 
+    artrepgen --sock 12345 >      .real &
+    ./usleep $repGenStartupTime
+
     cd "$d" && CFLAGS="${CFLAGS} -g -ggdb" RT_TEMPLATE=$RT_TEMPLATE rt-gmake \
     && ./main || true
-    artrepgen --file tracefile.out > .real
+
+    ./usleep $repGenShutdownTime
+#    cat tmp
+    killall -9 artrepgen || true
     ${SED} -r -i 's/[0-9A-Z]{16}//g' .real
+
     diff -u .real .right
     if [ $? -eq 0 ]; then
         rm .real
